tsinfer/inference.py

```diff
@@ -947,8 +947,12 @@ def run(self):
             logger.info(f"Starting build for {self.num_ancestors} ancestors")
             progress = self.progress_monitor.get("ga_generate", self.num_ancestors)
             a = np.zeros(self.num_sites, dtype=np.int8)
-            root_time = max(self.timepoint_to_epoch.keys()) + 1
-            ultimate_ancestor_time = root_time + 1
+            root_time = max(self.timepoint_to_epoch.keys())
+            if root_time < 1:
+                root_time = 1  # The normal case, when time==freq
+            else:
+                root_time += 1  # Could happen with user-specified times
+            ultimate_ancestor_time = root_time + 1 / self.num_ancestors
```

Comment:
Why the new 1/N fraction?

Reply:
What do you suggest using instead? I could use root_time+1, as before, which would mean in the usual case most nodes would be at times `[0..1)`, apart from the root (at time 1), and the ultimate ancestor (at time 2). This creates weird plots with all node times 0..1 apart from one at time 2.

If we use `np.nextafter`, the augment_ancestors functionality fails, as it adds 1 to all the times. So `1 / self.num_ancestors` seemed like a reasonable epsilon.